core/src/test/scala/sttp/tapir/ValidatorTest.scala

```diff
@@ -107,17 +107,50 @@ class ValidatorTest extends AnyFlatSpec with Matchers {
     v(0) shouldBe List(ValidationError.Custom(0, "X has to be greater than 5!"))
   }
 
-  it should "validate enum" in {
+  it should "validate coproduct enum" in {
     Validator.derivedEnumeration[Color](Blue) shouldBe empty
   }
 
+  it should "validate enum" in {
+    Validator.derivedEnumeration[ColorEnum](ColorEnum.Green) shouldBe empty
+  }
+
+  it should "not compile for enum with parameter" in {
+    assertDoesNotCompile("""
+      Validator.derivedEnumeration[ColorEnumWithParam](ColorEnumWithParam.Red) shouldBe empty
+    """)
+  }
+
+  it should "not compile for malformed coproduct enum" in {
+    assertDoesNotCompile("""
+      Validator.derivedEnumeration[InvalidColorEnum](InvalidColorEnum.Blue) shouldBe empty
+    """)
+  }
+
   it should "validate closed set of ints" in {
     val v = Validator.enumeration(List(1, 2, 3, 4))
     v.apply(1) shouldBe empty
     v.apply(0) shouldBe List(ValidationError.Primitive(v, 0))
   }
+
 }
 
 sealed trait Color
-case object Blue extends Color
-case object Red extends Color
+case object Blue  extends Color
+case object Red   extends Color
+
+enum ColorEnum { 
+  case Green  extends ColorEnum
+  case Pink   extends ColorEnum
```

Comment:
unfortunately we can't add `enum` here, as this will now fail to compile with scala 2. We need to create a `ValidatorTest2` (or sth like that ;) ) in the `scala-3` source tree